src/jdk.compiler/share/classes/com/sun/tools/javac/comp/ReflectMethods.java

```diff
@@ -1637,6 +1606,180 @@ public void visitSwitchExpression(JCTree.JCSwitchExpression tree) {
+                    pushBody(headCl, FunctionType.VOID);
+
+                    append(CoreOp._yield());
+                    bodies.add(stack.body);
+
+                    // Pop label
+                    popBody();
+                } else {
+                    throw unsupported(tree);
+                }
+
+                // Statements body
+                switch (c.caseKind) {
+                    case RULE -> {
+                        pushBody(c.body, actionType);
+                        if (c.body instanceof JCTree.JCBlock b) {
+                            toValue(b);
+                            if (!(b.stats.last() instanceof JCTree.JCBreak)) {
+                                append(CoreOp._yield()); // @@@ _break is also an option
+                            }
+                        }
+                        else if (c.body instanceof JCTree.JCStatement s) {
+                            toValue(s);
+                            if (!(s instanceof JCThrow)) {
+                                append(CoreOp._yield());
+                            }
+                        }
+                        bodies.add(stack.body);
+
+                        // Pop block
+                        popBody();
+                    }
+                    case STATEMENT -> {
+                        // @@@ Avoid nesting for a single block? Goes against "say what you see"
+                        // boolean oneBlock = c.stats.size() == 1 && c.stats.head instanceof JCBlock;
+                        pushBody(c, actionType);
+
+                        scan(c.stats);
+
+                        appendTerminating(c.completesNormally ?
+                                headCl instanceof JCTree.JCDefaultCaseLabel ? CoreOp::_yield : ExtendedOp::switchFallthroughOp
+                                : CoreOp::unreachable);
+
+                        bodies.add(stack.body);
+
+                        // Pop block
+                        popBody();
+                    }
+                };
+            }
+
+            // if enhanced switch and no default label
+            // enhanced: target type not byte, char short, int, String, or, has case pattern or case null
+            boolean enhancedSw = !List.of(JavaType.BYTE, JavaType.CHAR, JavaType.SHORT, JavaType.INT,
+                    JavaType.J_L_BYTE, JavaType.J_L_CHARACTER, JavaType.J_L_SHORT, JavaType.J_L_INTEGER,
+                    JavaType.J_L_STRING).contains(typeToTypeElement(tree.selector.type));
+            enhancedSw = enhancedSw && !isEnum(tree.selector.type.tsym);
+            enhancedSw = enhancedSw || tree.patternSwitch;
+            enhancedSw = enhancedSw || tree.cases.stream().anyMatch(c -> c.labels.stream().anyMatch(l -> {
+                return l instanceof JCTree.JCConstantCaseLabel ccl && ccl.expr instanceof JCLiteral literal && literal.value == null;
+            }));
+            if (enhancedSw && !tree.hasUnconditionalPattern) {
```

Comment:
Add a test that generates this part of the model?

Reply:
Are you asking if the tests cover this ?

Reply:
Yes.

Reply:
We do have tests that check we only add default when switch statement is enhanced and there is no implicit/explicit default. These tests can be found in SwitchStatementTest and are named: nonEnhancedSwStatNoDefault, enhancedSwStatNoDefault1, enhancedSwStatNoDefault2 and enhancedSwStatUnconditionalPattern.